spec/admin.mediawiki

```diff
@@ -45,4 +45,25 @@ possible.
 
 See also [[fundamentals.mediawiki/#Exchange_Variables|Exchange Variables]].
 
-...
+==Administration API==
+
+The server will provide an HTTP API for performing various adminstrative tasks.
+
+'''API Endpoints'''
+{|
+! path      !! description
+|-
+| /config   || the current DEX configuration. See [[fundamentals.mediawiki/#Configuration_Data_Request|Configuration Data Request]]
+|-
+| /accounts || lists information about all known accounts
+| -
+| /accounts/{accountID} || list information about a specific account
+| -
+| /markets  || display status information for all markets
+| -
+| /market/{marketID} || display status information for a specific market
+| -
+| /market/{marketID}/suspend || schedule a market suspension at the end of the current epoch
+| -
+| /unban/{accountID} || clear an account's penalties and re-enable trading
```

Comment:
Feelings about the /ban API endpoint?  This relates to the previously discussed penalization notification, which should probably be signed by the server and include details of the infraction(s).

Reply:
I was gonna chat with you and @JoeGruffins about that. I see the question is still sort of in the air at https://github.com/decred/dcrdex/pull/469#issuecomment-643605220. My gut says that we shouldn't even have the option. Is there a good reason?

Reply:
Unforeseen mischief, including hack attempts, etc.  There are already a myriad of ways an operator can DoS a user, so making this seamless just make sense to me.  If the issue is sorted out, the unban function exists.

Reply:
BTW, the ban/unban will probably need to be renamed to suspend/reinstate or similar, or at least the ban function will need a duration, maybe even option to disallow connection, when we figure out more advanced scoring and temporary suspension mechanics.

Reply:
Fair enough. Will just add `/ban` for now.

Reply:
I think with the current rules, ban is not so useful. Maybe only to re-ban someone you accidentally unbanned. But new rules may be added in the future, that can't be done programmatically, or that need to be retroactively imposed.

However, it could be abused by the operator. A normal exchange works on trading fees, so banning isn't in the best interest of the operator. But since the DEX is based on an initial fee, unless a user is vocal about it, banning has no impact on the operator. I don't know if this is an actual problem. I think we will find out when the DEX goes into production. Of course, even without this endpoint, an operator can just switch a value in the db and have the same effect. One possible solution is to try to make ban data public, if it is indeed a problem.

Reply:
> banning has no impact on the operator

This may not be true. I guess having low volume would not get you new users. So there may be enough incentive for this to not be abused. If I were banned for nothing, I certainly wouldn't pay the fee again, I'd find another DEX.

Reply:
I agree with all of that. My sort of half-baked philosophy on this is that if we need to manually ban someone, then our software isn't doing it's job. But I also recognize that there are bugs, and a learning curve, and that the ability to ban someone who skirts the rules might be critical at some point. Right now, I'm feeling like we should leave it, and just make it one of our goals that nobody ever has to use it.

Reply:
I think we should have it in the code and document it.  It could be hacked into the code anyway, the DB could be fiddled with, and firewalls employed for banning, so it should be a supported function.  I don't think it make sense to neuter the operator on principle, especially since there will be legit reasons to close an account like this.  Consider that a programmatic ban according to the rules also potentially creates a burden for the operator to demonstrate the broken rules.  Little changes with a ban function IMO.  If the operator closes accounts, their exchange is going to be on everyone's shit list.